src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4837,8 +4851,30 @@ private void PreVisitBottomUp(CallNode node, int argCountVisited, Scope scopeNew
                 var argTypes = args.Select(_txb.GetType).ToArray();
                 bool fArgsValid;
 
+                // This error container is used as temporary container so we can trap type mismatch kind of error for
+                // deferred (unknown) type args and validate all the errors were caused due to deferred(unknown) type.
+                var checkInvocationErrors = new ErrorContainer();
+
                 // Typecheck the invocation and infer the return type.
-                fArgsValid = func.HandleCheckInvocation(_txb, args, argTypes, _txb.ErrorContainer, out returnType, out var nodeToCoercedTypeMap);
+                fArgsValid = func.HandleCheckInvocation(_txb, args, argTypes, checkInvocationErrors, out returnType, out var nodeToCoercedTypeMap);
```

Comment:
>HandleCheckInvocation [](http://example.com/codeflow?start=34&length=21)

We have several other instances in this file calling HandleCheckInvocation.

Do they need to be updated too?

Reply:
We don't need to update them in current interation, we may need to update them when we extend support of deferred to functions with lamda args